src/box/memtx_tx.c

```diff
@@ -2032,6 +2143,18 @@ memtx_tx_history_add_insert_stmt(struct txn_stmt *stmt,
 	for (uint32_t i = 0; i < space->index_count; i++) {
 		struct tuple *next = directly_replaced[i];
 		struct tuple *succ = direct_successor[i];
+		/*
+		 * The tuple was inserted as new according to the operating
+		 * transaction and it's not excluded from the index. It's
+		 * possible that the operation replaced a tuple in the index,
+		 * but, if so, the tuple is not visible to the inserter.
+		 */
+		if (old_tuple == NULL && !tuple_excluded[i])
+			memtx_tx_handle_count_conflicts(space, add_story, i);
```

Comment:
It's not clear to me now why this required a separated condition, maybe other conflict detecting functions called below should be moved here too.